William's Homework

wmac001 opened the pull request:
Homework 1 Completed! 

- William Mac

Thanks!

wmac001 committed:
Add files via upload

NimaChartab commented on Homework/Homework 1_WM.ipynb:
```diff
@@ -0,0 +1,1736 @@
+    "    \n",
+    "    \n",
+    "    def Performance(self):\n",
+    "        from matplotlib import pyplot as plt\n",
```
It is better to import libraries outside the function at the beginning of the code.

NimaChartab commented on Homework/Homework 1_WM.ipynb:
```diff
@@ -0,0 +1,1736 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Set = Matrix_Set(A_WM,O_WM,5,10)"
```
Didn't you get an error for this part? It returns "Matrix dimensions do not match"

wmac001 replied:
i left out self.C in both functions when i rewrote it to implement the matrix multiplication. :( It runs fun once I replaced the C with self.C and initialized self.C =[]

NimaChartab commented on Homework/Homework 1_WM.ipynb:
```diff
@@ -0,0 +1,1736 @@
+   "source": [
+    "np.polyfit(N,Dialer,1)\n",
+    "\n",
+    "ynew=np.poly1d(np.polyfit(N,Dialer,1))\n",
```
We need to fit a logarithmic function for this part.

NimaChartab reviewed (commented):
Your grade for HW1 is 85/100. Please find my comments on your code.

wmac001 committed:
Add files via upload

Homework for William Mac

wmac001 commented:
Homework #2 Submitted! - William Mac

wmac001 committed:
Homework #3 Submitted

wmac001 commented:
Homework #3 Submitted

NimaChartab commented on Homework/Homework 2_WM.ipynb:
```diff
@@ -0,0 +1,697 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "D_T = D.transpose(2,0,1).reshape(-1,D.shape[1])\n"
```
Don't need this part.

NimaChartab commented on Homework/Homework 2_WM.ipynb:
```diff
@@ -0,0 +1,697 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "λ1_mode=np.argmax(P1)\n",
```
λ1_mode=P1[np.argmax(P1)]

NimaChartab commented on Homework/Homework 2_WM.ipynb:
```diff
@@ -0,0 +1,697 @@
+   "outputs": [],
+   "source": [
+    "λ1_mode=np.argmax(P1)\n",
+    "λ2_mode=np.argmax(P2)\n",
```
λ2_mode=P2[np.argmax(P2)]

NimaChartab commented on Homework/Homework 2_WM.ipynb:
```diff
@@ -0,0 +1,697 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "P_ = D.sum(axis=2)\n",
```
Please check the solution for getting marginalized posteriors.

NimaChartab reviewed (commented):
Your grade for homework 2 is 75. Please check the solution and let me know if you have any questions.

wmac001 committed:
Homework #4 Submitted

wmac001 commented:
Homework #4 Submitted

wmac001 committed:
Add files via upload

You will see some blank spaces in the notebook because my graphs are unable to populate. I have attached the html for those graphs. For some reason after I upgraded my anaconda, I was unable to do fig.show for those plots so I had to export them to html and set auto_open = True.

wmac001 commented:
Final Project Submitted
